plugin/pkg/timeouts/timeouts.go

```diff
@@ -0,0 +1,30 @@
+package timeouts
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// NewTimeoutFromArg returns a time.Duration from a configuration argument
+// (string) which has come from the Corefile. The argument has some basic
+// validation applied before returning a time.Duration.
+func NewTimeoutFromArg(arg string) (time.Duration, error) {
+	_, err := strconv.Atoi(arg)
+	if err == nil {
+		// If no time unit is specified default to seconds rather than
+		// GO's default of nanoseconds.
+		arg = arg + "s"
+	}
+
+	d, err := time.ParseDuration(arg)
+	if err != nil {
+		return time.Duration(0), fmt.Errorf("failed to parse timeout duration '%s'", arg)
```

Comment:
just a nitpick, unnecessary boxing of 0 to time.Duration (I think there are multiple occurrences of this unnecessary boxing)
```suggestion
		return 0, fmt.Errorf("failed to parse timeout duration '%s'", arg)
```